FlowerShop/Repositories/MerchandiseCategoryRepository.cs

```diff
@@ -0,0 +1,62 @@
+﻿using FlowerShop.Data;
+using FlowerShop.Interfaces;
+using FlowerShop.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace FlowerShop.Repositories
+{
+    public class MerchandiseCategoryRepository : IMerchandiseCategoryRepository
+    {
+        private readonly DataContext _dataContext;
+
+        public MerchandiseCategoryRepository(DataContext dataContext) 
+        {
+            _dataContext = dataContext;
+        }
+
+        public async Task<MerchandiseCategory> AddMerchandiseCategory(MerchandiseCategory merchandiseCategory)
+        {
+            _dataContext.MerchandiseCategories.Add(merchandiseCategory);
+            await _dataContext.SaveChangesAsync();
+            return merchandiseCategory;
+        }
+
+        public async Task<MerchandiseCategory?> DeleteMerchandiseCateogry(Guid id)
+        {
+            var merchandise = await GetById(id);
+            if (merchandise == null)
+                return null;
+            _dataContext.MerchandiseCategories.Remove(merchandise);
+            await _dataContext.SaveChangesAsync();
+            return merchandise;
+        }
+
+        public async Task<IEnumerable<MerchandiseCategory>> GetAll() => await _dataContext.MerchandiseCategories.ToListAsync();
+
+        public async Task<IEnumerable<MerchandiseCategory>> GetBy(string? name = null, Guid? parentCategory = null)
+        {
+            var query = _dataContext.MerchandiseCategories.AsQueryable();
+
+            if(!string.IsNullOrEmpty(name)) 
+                query = query.Where(m => m.Name != null && m.Name.ToLower() == name.ToLower());
```

Comment:
I know that that's how I did it as well, but now that I think about it we shouldn't compare if a name is equal but whether the name in the database contains the string from the query. (For example, if we search for a name: Orc we would still get Orchids)